tests/test_server.py

```diff
@@ -210,32 +219,117 @@ def test_get_wal_full_path(self, tmpdir):
         full_path = server.get_wal_full_path(wal_name)
         assert full_path == str(tmpdir.join("wals").join(wal_hash).join(wal_name))
 
+    @pytest.mark.parametrize(
+        "wal_info_files,expected_indices",
+        [
+            (
+                [
+                    create_fake_info_file("000000010000000000000003", 42, 43),
+                    create_fake_info_file("00000001.history", 42, 43),
+                    create_fake_info_file("000000020000000000000003", 42, 43),
+                    create_fake_info_file("00000002.history", 42, 43),
+                    create_fake_info_file("000000030000000000000005", 42, 43),
+                    create_fake_info_file("00000003.history", 42, 43),
+                ],
+                [1, 2, 3, 5],
+            ),
+            (
+                [
+                    create_fake_info_file("000000010000000000000003", 42, 43),
+                    create_fake_info_file("00000001.history", 42, 43),
+                    create_fake_info_file("000000020000000000000003", 42, 43),
+                    create_fake_info_file("000000020000000000000010", 42, 43),
+                    create_fake_info_file("00000002.history", 42, 43),
+                    create_fake_info_file("000000030000000000000005", 42, 43),
+                    create_fake_info_file("00000003.history", 42, 43),
+                ],
+                [1, 2, 3, 4, 6],
+            ),
+        ],
+    )
+    def test_get_required_xlog_files(self, wal_info_files, expected_indices, tmpdir):
+        # Prepare input string
+        walstring = ""
+        for i in wal_info_files:
+            walstring += i.to_xlogdb_line()
```

Comment:
Minor nitpick: I think using the name `wal_info` instead of `i` would be more expressive here.